java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java

```diff
@@ -75,18 +75,19 @@ public StringHashTableDictionary(int initialCapacity, float loadFactor) {
     this.capacity = initialCapacity;
     this.loadFactor = loadFactor;
     this.keyOffsets = new DynamicIntArray(initialCapacity);
-    initHashBuckets(initialCapacity);
+    initHashBuckets(false);
     this.threshold = (int)Math.min(initialCapacity * loadFactor, MAX_ARRAY_SIZE + 1);
   }
 
-  private void initHashBuckets(int capacity) {
-    DynamicIntArray[] buckets = new DynamicIntArray[capacity];
-    for (int i = 0; i < capacity; i++) {
+  private void initHashBuckets(final boolean reinit) {
+    final DynamicIntArray[] newBuckets =
+        (reinit) ? this.hashBuckets : new DynamicIntArray[this.capacity];
+    for (int i = 0; i < this.capacity; i++) {
       // We don't need large bucket: If we have more than a handful of collisions,
       // then the table is too small or the function isn't good.
-      buckets[i] = createBucket();
+      newBuckets[i] = createBucket();
```

Comment:
The DynamicIntArray-typed elements are re-created anyway. Just curious if we can see obvious gain.

Reply:
Hello. Thanks for the review. The purpose of this patch is to preserve the `buckets` array since this method simply repopulates the contents. There's no reason to throw away a 4KB array to simply replace it with a new one .